Formula/grafana.rb

```diff
@@ -0,0 +1,123 @@
+require "language/go"
+require "language/node"
+require "open3"
+
+class Grafana < Formula
+  desc "Gorgeous metric visualizations and dashboards for timeseries databases."
+  homepage "http://grafana.org"
+  url "https://github.com/grafana/grafana/archive/v3.0.4.tar.gz"
+  sha256 "f26a374326e64a8f83c57fdf916ba1c8524dd55002dfe628b4854d05fed3715a"
+
+  head "https://github.com/grafana/grafana.git"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+
+  def install
+    ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+    ENV["GOPATH"] = buildpath
+    grafana_path = buildpath/"src/github.com/grafana/grafana"
+    grafana_path.install Dir["*"]
+    grafana_path.install ".jscs.json", ".jsfmtrc", ".jshintrc", ".bowerrc"
+
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd grafana_path do
+      # Might do it differently for head vs. release
+      system %q(sed -i.bak 's/"sass-lint": "^1.6.0",/"sass-lint": "1.7.0",/;' package.json)
+      system "go", "run", "build.go", "setup"
+      system "go", "run", "build.go", "build"
+      system "npm", "install", *Language::Node.local_npm_install_args
+      system "npm", "install", "grunt-cli", *Language::Node.local_npm_install_args
+      system "node_modules/grunt-cli/bin/grunt", "build"
+    end
+
+    cp(grafana_path/"conf/sample.ini", grafana_path/"conf/grafana.ini")
+
+    bin.install grafana_path/"bin/grafana-cli"
+    bin.install grafana_path/"bin/grafana-server"
+    (etc/"grafana").mkpath
+    (var/"log/grafana").mkpath
+    (var/"lib/grafana").mkpath
+    (var/"lib/grafana/plugins").mkpath
```

Comment:
`mkpath` is an alias for `mkdir_p` so if you use `(var/"lib/grafana/plugins").mkpath` you don't need the line above creating `(var/"lib/grafana")`